internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1318,6 +1319,18 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 							Type:     pluginsdk.TypeBool,
 							Optional: true,
 						},
+						"revisions": {
+							Type:     pluginsdk.TypeSet,
+							Optional: true,
+							Computed: true,
+							MinItems: 1,
```

Comment:
This would be a breaking change for users that have this block set.

Reply:
Sorry, are you referring to MinItems? As in it would be a breaking change when in future this MinItems changes? And vice versa for MaxItems?

Reply:
I was referring to the `MinItems`, but my suspicion that it would be a breaking change was incorrect. Specifying a `MinItems` would have implied that the property must be set with at least one element in the list, but because it's `Optional` is appears to really only be a validation that's applied if the property has been set. Good to know.

Unfortunately adding this property is still going to be a breaking change since users will need to add this property to `ignore_changes` to prevent a diff when they upgrade to a version that will include this change.

From the sounds of it it also seems like this property might actually want to be `Required` instead of `Optional`?